### PR TITLE
add price extractor

### PR DESCRIPTION
**Pull request by ZwEin27**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    name = 'digPriceExtractor',
+    version = '0.1.0',
+    description = 'digPriceExtractor',
+    author = 'Lingzhe Teng',
+    author_email = 'zwein27@gmail.com',
+    url = 'https://github.com/ZwEin27/dig-price-extractor',
+    download_url = 'https://github.com/ZwEin27/dig-price-extractor',
+    packages = find_packages(),
+    keywords = ['price', 'extractor'],
+    install_requires=['digSparkUtil', 'digExtractor', 'inflection']
```
Comment (jasonslepicka) [inline setup.py]: This should not include digSparkUtil and should include the packages actually required

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+from setuptools import Extension,find_packages
+from os import path
+
+setup(
+    name = 'digPriceExtractor',
+    version = '0.1.0',
+    description = 'digPriceExtractor',
+    author = 'Lingzhe Teng',
+    author_email = 'zwein27@gmail.com',
+    url = 'https://github.com/ZwEin27/dig-price-extractor',
```
Comment (jasonslepicka) [inline setup.py]: These should be the usc-isi-i2 urls

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Babel==2.3.4
+decorator==4.0.10
+digExtractor>=0.1.8
+inflection==0.3.1
+jsonpath-rw==1.4.0
+jsonpath-rw-ext==1.0.0
```
Comment (jasonslepicka) [inline requirements.txt]: jsonpath-rw and jsonpath-rw-ext should not be in here.  can you make sure the rest of these requirements are necessary too?

File: digPriceExtractor/price_extractor.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        return copy.copy(self.metadata)
+
+    def set_metadata(self, metadata):
+        self.metadata = metadata
+        return self
+
+    def get_renamed_input_fields(self):
+        return self.renamed_input_fields
+
+    def set_renamed_input_fields(self, renamed_input_fields):
```
Comment (jasonslepicka) [inline digPriceExtractor/price_extractor.py]: This can be removed since it's inherited

File: digPriceExtractor/digpe/preprocessor.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        text = Preprocessor.replace_numeral_words(text)
+        text = Preprocessor.re_irrelation.sub(' ', text)
+        text = Preprocessor.re_single_space.sub('', text)
+        text = Preprocessor.re_phone_number.sub('', text)
+        text = text.split('\n')
+        text = [' '.join([inflection.singularize(token) for token in _.split(' ')]) for _ in text]
+        return text
+        # return text.split('\n') # future find all instead
+
+if __name__ == '__main__':
```
Comment (jasonslepicka) [inline digPriceExtractor/digpe/preprocessor.py]: This should be removed

File: digPriceExtractor/digpe/preprocessor.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    def preprocess(self, text):
+        text = text.encode('ascii', 'ignore').lower()
+        text = Preprocessor.replace_numeral_words(text)
+        text = Preprocessor.re_irrelation.sub(' ', text)
+        text = Preprocessor.re_single_space.sub('', text)
+        text = Preprocessor.re_phone_number.sub('', text)
+        text = text.split('\n')
+        text = [' '.join([inflection.singularize(token) for token in _.split(' ')]) for _ in text]
+        return text
+        # return text.split('\n') # future find all instead
```
Comment (jasonslepicka) [inline digPriceExtractor/digpe/preprocessor.py]: Commented out code should be removed.

File: digPriceExtractor/digpe/normalizer.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        return ht
+
+
+    def normalize_from_list(self, text_list):
+        return [self.normalize(text) for text in text_list]
+
+
+
+if __name__ == '__main__':
+    # text = 'half hour 100'
```
Comment (jasonslepicka) [inline digPriceExtractor/digpe/normalizer.py]: This should be removed or converted into unit tests

File: digPriceExtractor/digpe/extractor.py [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+time: hh, hr...
+unit: dollar, rose, candy...
+
+
+"""
+import re
+
+from unit import *
+
+class Extractor():
```
Comment (jasonslepicka) [inline digPriceExtractor/digpe/extractor.py]: We may want to rename this to avoid collisions/confusion

File: digPriceExtractor/digpe/__init__.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+
+PE_DICT_NAME_PRICE = 'price'
+PE_DICT_NAME_PPH = 'price_per_hour'
+
+PE_JSON_NAME_PRICE = 'price'
+PE_JSON_NAME_PRICE_UNIT = 'price_unit'
+PE_JSON_NAME_TIME_UNIT = 'time_unit'
+
+class DIGPE():
```
Comment (jasonslepicka) [inline digPriceExtractor/digpe/__init__.py]: We should rename this to be a little more informative.

File: digPriceExtractor/digpe/__init__.py [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+            ans[PE_DICT_NAME_PRICE].append(normalized)
+        return ans
+
+
+    def extract_from_list(self, text_list):
+        return [self.extract(text) for text in text_list]
+
+if __name__ == '__main__':
+    # text = '$550 1 hr'
```
Comment (jasonslepicka) [inline digPriceExtractor/digpe/__init__.py]: This should be removed or turned into unit tests

**Review by jasonslepicka — CHANGES_REQUESTED** [review]
Looks good.  Just needs some cleanup

**Comment by ZwEin27** [comment]
fixed issues, thank you,:)

**Review by jasonslepicka — APPROVED** [review]

**Pull request merged**